galsim/celestial.py

```diff
@@ -582,7 +585,79 @@ def galactic(self, epoch=2000.):
 
         return (el, b)
 
+    def ecliptic(self, epoch=2000., julian_date=None):
+        """Get the longitude and latitude in ecliptic coordinates corresponding to this position.
+
+        Technically, this routine by default returns the ecliptic coordinates with respect to those
+        of the sun.  Equivalently one can think of it as returning the ecliptic coordinates at the
+        vernal equinox, when the sun is located at (0,0) in ecliptic coordinates.  The `epoch`
+        parameter is used to get an accurate value for the (time-varying) obliquity of the ecliptic,
+        but not to account for the sun's varying position in ecliptic coordinates as a function of
+        time.  To get an absolute value of the ecliptic coordinates for some arbitrary Julian date
+        taking into account the sun position on that date, use the `julian_date` argument.
+
+        The formulae for this are quite straightforward.  It requires just a single parameter for
+        the transformation, the obliquity of the ecliptic (the Earth's axial tilt).
+
+        @param  epoch        The epoch to be used for estimating the obliquity of the ecliptic.
+                             [default: 2000.]
+        @param  julian_date  The Julian date to be used to calculate the ecliptic coordinates.  If
+                             None, then the ecliptic coordinates that are returned should be
+                             interpreted as those with respect to the sun (i.e., assuming the sun is
+                             at (0,0) as it is at the vernal equinox).
+                             [default: None]
```

Comment:
This is backwards.  If `julian_date is None`, then this routine returns the [true ecliptic coordinates](http://en.wikipedia.org/wiki/Ecliptic_coordinate_system).  If you give it a date, then it gives you the coordinates with respect to the sun being at (0,0).

Also, python has a [datetime](https://docs.python.org/2/library/datetime.html) module.  Perhaps we ought to use that for passing around the date.  

[Here](http://code-highlights.blogspot.com/2013/01/julian-date-in-python.html) is code to convert from that to the julian date.  But I'm not sure how many people will find the raw julian date value very useful.